doc/api/esm.md

```diff
@@ -156,7 +156,7 @@ typically configured server.
 ### URLs
 
 ES modules are resolved and cached as URLs. This means that files containing
-special characters such as `#` and `?` need to be escaped.
+special characters need to be replaced with escape sequences, such as `#` with `%23` and `?` with `%3F`.
```

Comment:
```suggestion
special characters must be [percent-encoded][], such as `#` with `%23` and `?` with `%3F`.
```

Then add a new reference at the bottom of the esm.md to:

```markdown
[percent-encoded]: url.md#percent-encoding-in-urls
```

Reply:
Maybe either the bit being changed or the `percent-encoding-in-urls` section of `url.md` could point to [`encodeURIComponent()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)?

Reply:
> Maybe either the bit being changed or the `percent-encoding-in-urls` section of `url.md` could point to [`encodeURIComponent()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)?

Or https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding?